public/calendar_reboot.js

```diff
@@ -808,32 +826,63 @@ Calendar.prototype.fillCalendar = function(monthObjectsArray) {
 var LocalCalendarStorage = function(params) {
     var self = this;
     var prefix = params['storeId'] || "";
+    var event = params['event'];
     var allCalendarIdsKey = 'allCalendarIdsKey';
     //the current_active_calendar is the key for localStorage that stores
     //the active calendar's Id
     var current_active_calendar = 'current_active_calendar';
+    // This tells you whether the storage is actively working.
+    self.activeCalls = 0;
     
-    var toKey = function(id) {
+     var toKey = function(id) {
         //make a key out of a uniqueId
         
         var key = prefix + "_" + id;
         return key;
     };
     
+    self.startWork = function() {
+        // Will increment the counter and possibly fire an event.
+        console.log("Accessing storage");
+        self.activeCalls += 1;
+        
+        // Will dispatch the event backgroundActivityChange 
+        console.log("dipatching backgroundActivityChange");
```

Comment:
I think you only want to do this when `activeCalls` is nonzero. The event handler will have no way of knowing this the way it is currently written.

You *could* include the current value of `activeCalls` as data on the event object, but I don't think that's necessary for what you're trying to do.